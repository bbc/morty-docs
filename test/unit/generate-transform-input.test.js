const generateTransformInput = require('../../src/generate-transform-input')
const readdir = require('recursive-readdir')
const fs = require('fs')

jest.mock('recursive-readdir')
jest.mock('fs')

readdir.mockImplementation(() => Promise.resolve(['someRepo/someDirectory/some/path/tofile.md', 'someRepo/someDirectory/another/path/tofile.md']))
fs.readFileSync.mockImplementation(() => 'some text in a file')

describe('Generate transform input', () => {
  it('creates a correctly structured array of file objects from a directory ending in a slash', async () => {
    const expected = [
      {
        relativePath: 'someRepo/someDirectory/some/path/tofile.md',
        raw: 'some text in a file'
      },
      {
        relativePath: 'someRepo/someDirectory/another/path/tofile.md',
        raw: 'some text in a file'
      }
    ]

    const actual = await generateTransformInput('someRepo/someDirectory/')

    expect(actual).toEqual(expected)
  })

  it('creates a correctly structured array of file objects from a directory that does not end in a slash', async () => {
    const expected = [
      {
        relativePath: 'someRepo/someDirectory/some/path/tofile.md',
        raw: 'some text in a file'
      },
      {
        relativePath: 'someRepo/someDirectory/another/path/tofile.md',
        raw: 'some text in a file'
      }
    ]

    const actual = await generateTransformInput('someRepo/someDirectory')

    expect(actual).toEqual(expected)
  })
})
