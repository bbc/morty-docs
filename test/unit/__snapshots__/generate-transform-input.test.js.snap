// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test Files in folder \`test1\`, should match snapshot with \`1\` files 1`] = `
Array [
  Object {
    "raw": Object {
      "data": Array [
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "test1.txt",
  },
]
`;

exports[`Snapshot Test Files in folder \`test2\`, should match snapshot with \`2\` files 1`] = `
Array [
  Object {
    "raw": Object {
      "data": Array [
        68,
        105,
        114,
        32,
        49,
        32,
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "dir1/test1.txt",
  },
  Object {
    "raw": Object {
      "data": Array [
        68,
        105,
        114,
        32,
        50,
        32,
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "dir2/test1.txt",
  },
]
`;

exports[`Snapshot Test Files in folder \`test3\`, should match snapshot with \`3\` files 1`] = `
Array [
  Object {
    "raw": Object {
      "data": Array [
        68,
        105,
        114,
        32,
        49,
        32,
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "dir1/test1.txt",
  },
  Object {
    "raw": Object {
      "data": Array [
        68,
        105,
        114,
        32,
        49,
        32,
        83,
        117,
        98,
        32,
        49,
        32,
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "dir1/sub1/test1.txt",
  },
  Object {
    "raw": Object {
      "data": Array [
        68,
        105,
        114,
        32,
        49,
        32,
        83,
        117,
        98,
        32,
        50,
        32,
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "dir1/sub1/sub2/test1.txt",
  },
]
`;

exports[`Snapshot Test Files in folder \`test4\`, should match snapshot with \`2\` files 1`] = `
Array [
  Object {
    "raw": Object {
      "data": Array [
        83,
        121,
        109,
        108,
        105,
        110,
        107,
        32,
        84,
        101,
        120,
        116,
      ],
      "type": "Buffer",
    },
    "relativePath": "symlink.txt",
  },
  Object {
    "raw": Object {
      "data": Array [
        68,
        105,
        114,
        32,
        49,
        32,
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "dir1/test1.txt",
  },
]
`;

exports[`Snapshot Test Files in folder \`test5\`, should match snapshot with \`1\` files 1`] = `
Array [
  Object {
    "raw": Object {
      "data": Array [
        83,
        121,
        109,
        108,
        105,
        110,
        107,
        32,
        84,
        101,
        120,
        116,
      ],
      "type": "Buffer",
    },
    "relativePath": "symlink.txt",
  },
]
`;

exports[`Snapshot Test Files in folder \`test6\`, should match snapshot with \`0\` files 1`] = `Array []`;

exports[`Snapshot Test should match test1 snapshot 1`] = `
Array [
  Object {
    "raw": Object {
      "data": Array [
        84,
        101,
        115,
        116,
        32,
        49,
      ],
      "type": "Buffer",
    },
    "relativePath": "test1.txt",
  },
]
`;
